Jim-Hodapp-Coaching/refactor-platform-rs
# axum-login credential-checking and user session creation

## Description
Adds axum-login based credential-checking and user session creation/checking/deletion available at two specific new endpoints:

`/login`
`/logout`


#### GitHub Issue: None

### Changes
* Adds a password field to the `User` entity. Make sure to rerun DB migrations.
* Protects all existing entity-api endpoints behind needing to log in and create a session
* Allows 2 DB-seeded users to log in and can maintain the 2 distinct sessions
* The entity-api endpoints do work, e.g. organizations/*, but only if you pass the session id with every request
* The user can log out thus destroying the session
* Uses PostgresStore to store created user sessions, currently stores them under `tower_sessions/sessions`

### Testing Strategy
* Log in: `curl -v --header "Content-Type: application/x-www-form-urlencoded" --data "username=james.hodapp@gmail.com&password=password1&next=organizations" http://localhost:4000/login`
* Notice the cookie that is create and mentioned in the reply which includes a session ID. Copy and replace <session_id> for all other requests.
* Test an organization endpoint: `curl -v --header "Cookie: id=<session_id>" --request GET http://localhost:4000/organizations`
* This should work as expected
* Log out: `curl -v --header "Cookie: id=<session_id>" --request GET http://localhost:4000/logout`


### Concerns
- The existing endpoints tests are broken, they need to be updated to log in and create/use a session
